### PR TITLE
update to latest schema refactors, e.g. status codes, responses, etc

### PR DESCRIPTION
**Pull request by vpchung**
With this PR:

- update to use latest schema changes, e.g. new models, updated status codes, etc.

TODO:
- [x] fix tests to reflect status code updates
- [x] add tests to increase coverage
- [x] fix unintended `500 Internal Error` for when a value is not found in the database (should be `404 Not Found`)

### CODE REVIEW

**Comment by vpchung** [comment]
Converting to draft for now; I want to add/update the tests so that the checks are successful.

File: server/openapi_server/controllers/challenge_controller.py [hunk 1]
```diff
@@ -44,30 +43,30 @@ def create_challenge(challenge=None):
...
+                    # grant=challenge.grant,
+                    organizers=challenge.organizers,
+                    tags=challenge.tags,
+                    challengeResults=DbChallengeResults(
+                        nSubmissions=challenge.challenge_results.n_submissions,  # noqa: E501
+                        nFinalSubmissions=challenge.challenge_results.n_final_submissions,  # noqa: E501
+                        nRegisteredParticipants=challenge.challenge_results.n_registered_participants,  # noqa: E501
+                    )
+                ).save(force_insert=True)
+                res = ChallengeCreateResponse.from_dict(db_challenge.to_dict())
```
Comment (tschaffter) [inline server/openapi_server/controllers/challenge_controller.py:65]: Does this return a payload that has only the property `challengeId`? If yes, this line makes an assumption that the properties of `ChallengeCreateResponse` (there's actually only one property) are a subset of the properties of `Challenge`. This assumption may lead to unforeseen effect in the future. The code would be more robust if we create the object `ChallengeCreateResponse` using its constructor.

**Review by tschaffter — CHANGES_REQUESTED** [review]
Just one comment to address

File: server/openapi_server/controllers/challenge_controller.py (hunk 1 above)
Comment (vpchung) [reply server/openapi_server/controllers/challenge_controller.py:65]: @tschaffter yes, the returned payload contains just the `challengeId`:

Screen Shot 2021-01-21 at 5 28 18 PM

> This assumption may lead to unforeseen effect in the future. The code would be more robust if we create the object `ChallengeCreateResponse` using its constructor.

Thanks for the insight! I didn't think of that.

**Review by tschaffter — APPROVED** [review]
Looks great! Nicely done!

**Pull request merged**